qiskit_experiments/library/randomized_benchmarking/interleaved_rb_experiment.py

```diff
@@ -108,60 +155,55 @@ def circuits(self) -> List[QuantumCircuit]:
         Returns:
             A list of :class:`QuantumCircuit`.
 
+        Raises:
+            QiskitError: if failed to transpile interleaved_element.
         """
-        if self.num_qubits > 2:
-            return super().circuits()
-
-        self._set_basis_gates()
-        self._initialize_clifford_utils()
-        rng = default_rng(seed=self.experiment_options.seed)
-        circuits = []
-        for _ in range(self.experiment_options.num_samples):
-            self._set_transpiled_interleaved_element()
-            std_circuits, int_circuits = self._build_rb_circuits(
-                self.experiment_options.lengths,
-                rng,
-            )
-            circuits += std_circuits
-            circuits += int_circuits
-        return circuits
+        basis_gates = self._get_basis_gates()
+        self._cliff_utils = CliffordUtils(self.num_qubits, basis_gates=basis_gates)  # TODO: cleanup
 
-    def _set_transpiled_interleaved_element(self):
-        """
-        Create the transpiled interleaved element. If it is a single gate,
-        create a circuit comprising this gate.
-        """
+        # Convert interleaved element to transpiled circuit operations and store them for speed
+        # Convert interleaved element to circuit
         if isinstance(self._interleaved_op, QuantumCircuit):
-            qc_interleaved = self._interleaved_op
-        else:
-            qc_interleaved = QuantumCircuit(self.num_qubits, self.num_qubits)
-            qubits = list(range(self.num_qubits))
-            qc_interleaved.append(self._interleaved_op, qubits)
-            self._transpiled_interleaved_elem = qc_interleaved
-
-        if hasattr(self.transpile_options, "basis_gates"):
-            basis_gates = self.transpile_options.basis_gates
-        else:
-            basis_gates = None
-        self._transpiled_interleaved_elem = transpile(
-            circuits=qc_interleaved,
-            optimization_level=1,
-            basis_gates=basis_gates,
-            backend=self._backend,
-        )
-
-    def _sample_circuits(self) -> List[QuantumCircuit]:
-        """Return a list of RB circuits.
+            interleaved_circ = self._interleaved_op
+        elif isinstance(self._interleaved_op, Clifford):
+            interleaved_circ = self._interleaved_op.to_circuit()
+        elif isinstance(self._interleaved_op, Gate):
+            interleaved_circ = QuantumCircuit(self.num_qubits, name=self._interleaved_op.name)
+            interleaved_circ.append(self._interleaved_op, list(range(self.num_qubits)))
+        else:  # Delay
+            interleaved_circ = []
+        if basis_gates and any(i.operation.name not in basis_gates for i in interleaved_circ):
+            interleaved_circ.name = f"Clifford-{interleaved_circ.name}"
+            # Transpile circuit with non-basis gates and remove idling qubits
+            try:
+                interleaved_circ = transpile(
```

Comment:
Do you need full transpile here? Likely you just need to call basis translator.

Reply:
Good point. I think it was for delays but it may be unnecessary now since I updated the handling of delays in [36193c9](https://github.com/Qiskit/qiskit-experiments/pull/922/commits/36193c94e635a1bdef65b97acf573ea08d109c31)
Let me check if we can replace transpile with basis translation.

Reply:
I found we could but it would be difficult to explain the spec in doc (since we need to mention to a protected method `_transform_clifford_circuit`). Now I think we should revisit here when we redesign the basis translation scheme (removing the use of `transpile` for 1Q/2Q RB) in the future.